### PR TITLE
Add imu support and vesc udev enumeration tools

### PR DESCRIPTION
**Pull request by anscipione**
- added tool to enumerate vescs based on their uuid (/dev/vesc/XXXX-XXXX-XXXX)
- added support to read imu data

### CODE REVIEW

File: vesc_driver/package.xml [hunk 1]
```diff
@@ -21,7 +21,9 @@
   <depend>rclcpp</depend>
   <depend>rclcpp_components</depend>
   <depend>std_msgs</depend>
+  <depend>geometry_msgs</depend>
   <depend>vesc_msgs</depend>
+  <depend>sensor_msgs</depend>
```
Comment (JWhitleyWork) [inline vesc_driver/package.xml:26]: It looks like you intended to publish `sensor_msgs/msg/Imu`. Can you please add this functionality? This is significantly more helpful than a VESC-specific message.

File: vesc_driver/src/vesc_interface.cpp [hunk 2]
```diff
@@ -161,9 +161,8 @@ void VescInterface::Impl::rxThread()
         error_handler_("Possibly out-of-sync with VESC, read timout in the middle of a frame.");
       }
     }
-
     // Only attempt to read every 10 ms
-    std::this_thread::sleep_for(std::chrono::milliseconds(10));
+//    std::this_thread::sleep_for(std::chrono::milliseconds(5));
```
Comment (JWhitleyWork) [inline vesc_driver/src/vesc_interface.cpp:165]: Can you please explain this change?

**Review by JWhitleyWork — CHANGES_REQUESTED** [review]
Awesome work! Just a couple of requests/suggestions.

File: vesc_driver/src/vesc_interface.cpp (hunk 2 above)
Comment (anscipione) [reply vesc_driver/src/vesc_interface.cpp:165]: I tried various waiting times but, also introducing the reading of IMU data, any waiting time produced a loss of data.  removing the wait solved the problem.

I'll remove this comments.

File: vesc_driver/package.xml (hunk 1 above)
Comment (anscipione) [reply vesc_driver/package.xml:26]: correct.  I wanted to post a standard message.  but, to remain more adherent to the data that the controller emits, I decided to define a message that would report all the data present.  (especially for debugging).  I can change it (orientation data in rpy format and magnetometer data are lost).  the magnetometer on mine are always at 0

Comment (JWhitleyWork) [reply vesc_driver/package.xml:26]: Sorry for the long delay. `sensor_msgs/msg/Imu` is a standard message and lots of stacks will accept this message. Please either change to publishing this message or at least add it as a second output.

**Comment by JWhitleyWork** [comment]
@anscipione Can you please address the issue I mentioned above and `rebase` your branch on `ros2`?

**Comment by anscipione** [comment]
yes I start working next week after returning from vacation

**Comment by anscipione** [comment]
> 
> 
> @anscipione Can you please address the issue I mentioned above and `rebase` your branch on `ros2`?

are you sure i should perform a rabase? if I have to proceed in this sense I have to close this pool request and open a new one with the rebase

**Comment by anscipione** [comment]
I opened a pull request with the rebase but the checks do not pass because the code on which I have based the rebase lacks the copyright.

**Comment by JWhitleyWork** [comment]
Closing. Hoping for a new PR against the current `ros2` branch.

**Pull request closed without merging**